browser/js/content.js

```diff
@@ -75,28 +81,67 @@ function renderChat(parent, toUsername) {
     <p>Encrypt to <span class="keybase-username">'+ toUsername +'</span>:</p>\
     <p><textarea name="keybase-chat" rows="6"></textarea></p>\
     '+ nudgeHTML +'\
-    <p><input type="submit" value="Send" /></p> \
+    <p><input type="submit" value="Send" name="keybase-submit" /></p> \
   ';
   f.addEventListener("submit", submitChat);
   parent.insertBefore(f, parent.firstChild);
 
+  // Install nudge toggle
+  const nudgeCheck = f["keybase-nudge"];
+  if (nudgeCheck !== undefined) {
+    // Select the <p><textarea>...</textarea></p>
+    const nudgeText = nudgeCheck.parentNode.parentNode.nextElementSibling;
+    nudgeCheck.addEventListener("change", function(e) {
+      nudgeText.hidden = !e.currentTarget.checked;
+    });
+  }
+
   // Install closing button (the "x" in the corner)
   const closer = f.getElementsByClassName("keybase-close")[0];
   closer.addEventListener("click", function(e) {
-    parent.removeChild(f);
+    removeChat(f);
   });
+
+  // TODO: Also add an onbeforeunload check if chat has text written in it.
+}
+
+// Remove the chat widget from the DOM
+function removeChat(chatForm, skipCheck) {
+  if (!skipCheck && chatForm["keybase-chat"].value != "") {
```

Comment:
I'd consider it a breaking bug if removeChat is called twice, or passes an invalid value (garbage in, garbage out). I added a check for the parent just in case though.